### ISSUE TITLE
Android: Using setValue() with a text string that has more than 2 spaces in it won't set the value

### ISSUE THREAD

**Issue by ttisdon** [appium/java-client#564]
## Description

Using setValue() with a text string that has more than 2 spaces in it won't actually set the value.

## Environment

* java client build version or git revision if you use some shapshot: 4.1.2
* Appium server version or git revision if you use some shapshot: 1.6.3
* Desktop OS/version used to run Appium if necessary: Mac OSX 10.11.16
* Node.js version (unless using Appium.app|exe) or Appium CLI or Appium.app|exe:
* Mobile platform/version under test: Android
* Real device or emulator/simulator: Emulator (Galaxy S5)

## Details
I'm testing a native app on the Galaxy S5 emulator (I made it with device definitions and such with Android AVD manager). I'm trying to setValue() the string of text "TESTING ALL THINGS ANDROID" into an android.widget.EditText field. When the test runs, the space gets highlit then no text is entered. I believe it has to do with ADB not putting a %s for each space.
I can setValue() text like "ANDROID TEST" or "TESTING" and the issue doesn't occur.

## Code To Reproduce Issue [ Good To Have ]

```
AppiumDriver<MobileElement> androidDriver;

File androidApp = new File(applicationUnderTest);
DesiredCapabilities androidVirtualNativeCapabilities = new DesiredCapabilities();
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.FULL_RESET, true);
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "Android Emulator");
androidVirtualNativeCapabilities.setCapability(AndroidMobileCapabilityType.AVD, virtualDevice);
androidVirtualNativeCapabilities.setCapability(AndroidMobileCapabilityType.AVD_ARGS, String.format("-wipe-data -no-boot-anim -port %s", emulatorPort));
androidVirtualNativeCapabilities.setCapability(AndroidMobileCapabilityType.APP_WAIT_ACTIVITY, androidActivity);
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.APP, androidApp.getAbsolutePath());
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.NEW_COMMAND_TIMEOUT, newCommandTimeout);
androidDriver = new AndroidDriver<>(new URL(appiumServerURL), androidVirtualNativeCapabilities);
androidDriver.context("NATIVE_APP");

@AndroidFindBy(xpath = "//android.widget.EditText[contains(@resource-id, 'add_contact_note_title')]")
private MobileElement ADD_TITLE_FIELD; // Title field

public void type(MobileElement mobileElement, String data) {
        mobileElement.setValue(data);
}

public void testingThisIssue() {
androidDriver.type(ADD_TITLE_FIELD, "TESTING ALL THINGS ANDROID");
}
```

## Exception stacktraces

There isn't a stacktrace. The test passes but the value isn't set. I believe the problem is with ADB. Here is a snippet from the Appium logs:

`[ADB] Running '/Users/trumaine.tisdon/android-sdk-macosx/platform-tools/adb' with args: ["-P",5037,"-s","emulator-5554","shell","input","text","TESTING%sALL THINGS ANDROID"]`

ADB puts a %s for the first space, but not for the other spaces and I think it's messing things up.

## Link to Appium logs

https://gist.github.com/ttisdon/2bfc4c3c8b3923f3c3f7d7b1a7c06e73#file-gistfile1-txt

**Comment by saikrishna321** [comment]
@ttisdon can you pls try using UIAutomator2 and see if the issue is still there

**Comment by ttisdon** [comment]
@saikrishna321 how should I run UIAutomator2?

**Comment by SrinivasanTarget** [comment]
```capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, AutomationName.ANDROID_UIAUTOMATOR2);```

**Comment by ttisdon** [comment]
I'm using Java Client 5.0.0-BETA1 and here are my capabilities:

```
AppiumDriver<MobileElement> androidDriver;

File androidApp = new File(applicationUnderTest);
DesiredCapabilities androidVirtualNativeCapabilities = new DesiredCapabilities();
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.FULL_RESET, true);
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "Android Emulator");
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, AutomationName.ANDROID_UIAUTOMATOR2);
androidVirtualNativeCapabilities.setCapability(AndroidMobileCapabilityType.AVD, virtualDevice);
androidVirtualNativeCapabilities.setCapability(AndroidMobileCapabilityType.AVD_ARGS, String.format("-wipe-data -no-boot-anim -port %s", emulatorPort));
androidVirtualNativeCapabilities.setCapability(AndroidMobileCapabilityType.APP_WAIT_ACTIVITY, androidActivity);
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.APP, androidApp.getAbsolutePath());
androidVirtualNativeCapabilities.setCapability(MobileCapabilityType.NEW_COMMAND_TIMEOUT, newCommandTimeout);
androidDriver = new AndroidDriver<>(new URL(appiumServerURL), androidVirtualNativeCapabilities);
androidDriver.context("NATIVE_APP");
```

When I start up the server and run my test, the app loads but then fails with this error:
`java.lang.NullPointerException: null value in entry: deviceScreenSize=null`

should I try a newer beta version?

**Comment by ttisdon** [comment]
I tried this recently with:
Android Studio 2.2
Java Client 5.0.0-BETA8
Selenium 3.4.0
UiAutomator 2

The issue is still happening. I tried to input the text FIRSTNAME WITH LOTS OF SPACES in an edit text field and this is what the logs say:

https://gist.github.com/ttisdon/6830d4af1d3062b1c7033671606b0a35

The last line says:
`[debug] [ADB] Running '/Users/trumaine.tisdon/Library/Android/sdk/platform-tools/adb' with args: ["-P",5037,"-s","emulator-5554","shell","input","text","FIRSTNAME%sWITH LOTS OF SPACES"]`

**Comment by mykola-mokhnach** [comment]
This is issue has recently been fixed in Appium 1.6.5-beta

**Comment by TikhomirovSergey** [comment]
So update of the server should resolve the problem.
